packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -496,6 +565,30 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   String toString() => '${objectRuntimeType(this, 'TransitionRoute')}(animation: $_controller)';
 }
 
+/// An interface for a route that supports predictive back gestures.
+///
+/// See also:
+///
+///  * [PredictiveBackPageTransitionsBuilder], which builds page transitions for
+///    predictive back.
+mixin PredictiveBackRoute {
```

Comment:
why is this a mixin while the api docs says "an interface"?

Reply:
(also the mixin isn't providing any implementations it seems?)

Reply:
Yeah, I think this should actually be `abstract interface`, and then it should be used with `implements`.